src/python/pants/backend/sql/target_types.py

```diff
@@ -0,0 +1,77 @@
+from typing import ClassVar
+
+from pants.core.target_types import FileSourceField, ResourceSourceField
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    Dependencies,
+    MultipleSourcesField,
+    OverridesField,
+    SingleSourceField,
+    StringField,
+    Target,
+    TargetFilesGenerator,
+    TargetFilesGeneratorSettingsRequest,
+    generate_file_based_overrides_field_help_message,
+    generate_multiple_sources_field_help_message,
+)
+from pants.util.strutil import help_text
+
+
+class SqlSourceField(ResourceSourceField):
+    """
+    Subclassing ResourceSourceField will make sure the sql is included in
+    distributions as a resource.
+    """
+
+    expected_file_extensions: ClassVar[tuple[str, ...]] = (".sql",)
```

Comment:
Is the file extension actually a hard rule?
Having this makes it impossible to use any other extension.. which seems bad if the tools used on these files are agnostic to file extensions? (such as the `sql_file_exts` config option for sqlfluff couldn't be used.)

https://github.com/pantsbuild/pants/blob/06517c85e3b5d30defd6120708f2cdd5842210db/src/python/pants/engine/target.py#L2088-L2089

Reply:
Yeah, this makes sense, removed